contributor-docs/deprecating-components.md

```diff
@@ -0,0 +1,78 @@
+
+### Code side of the deprecation
+
+Here is a checklist for developers to smoothen the deprecation process.
+
+#### Source Code
+
+- [ ] Move the old component's source code under the `src/deprecated` folder.
+- [ ] Remove the old component export in the main bundle to be added to the deprecated one.
+- [ ] Export the old component in the deprecated bundle `src/deprecated/index.ts`.
+- [ ] Rename the new component to its neutral version if it includes `2` or `v2`. I.e. `ActionList2` -> `ActionList`.
+- [ ] Move the new component's source code to the `src` folder if it is under `src/drafts`.
+- [ ] Export the new component in the main bundle `src/index.ts`.
+
+#### Storybook
+
+- [ ] Update the new component's storybook title from `Drafts/Components/<Component-Name>` to `Components/<Component-Name>`.
+- [ ] Move the old component's stories under `src/stories/deprecated`.
+
+#### Tests
+
+- [ ] Move the old component's tests under `src/__tests__/deprecated`.
+- [ ] Update `script/generate-e2e-tests.js` to reflect the new component's component ID is generated by the storybook title.
+- [ ] Update the snapshots using the command `npm run test -- -u`.
+
+#### Docs
+
+- [ ] Move the deprecated component's docs to the `src/docs/deprecated` folder and update the title by adding `(legacy)`, status as well as the storybook and the source code link.
+  - [ ] Make sure to update `jsx live` -> `jsx live deprecated`
+  - [ ] Make sure to update the import code block. I.e. `import { ActionList } from "@primer/react/deprecated"`.
+- [ ] Add a `Deprecation` section with the link of the newly developed component and provide a diff. See [deprecated ActionList docs](https://primer.style/react/deprecated/ActionList#deprecation) as an example.
+- [ ] Move the new components docs from draft folder to the main docs folder `docs/content/` and update the title by removing `v2`, status as well as the storybook and the source code link.
+  - [ ] Make sure to update `jsx live drafts` -> `jsx live`
+  - [ ] Make sure to update the import code block. I.e. `import { ActionList } from "@primer/react"`.
+  - [ ] Update the navigation on `docs/src/@primer/gatsby-theme-doctocat/nav.yml` accordingly.
+
+### Comms side of the deprecation
+
+When we know if the newly developed component is ready to go in the main bundle and the old component is good to be deprecates in the next major release, we announce the deprecation and the promotion in the Primer changelog under the `Coming soon` section.
+
+Once the major release is out, we provide support on how to switch using the newly develop component as well as how to continue to use the deprecated component for folks who prefer to migrate later on.
+
+We also write codemods to deprecate the old component and use the new component under [our migration repo](https://github.com/primer/react-migrate#readme).
```

Comment:
Do we provide any other support for teams except proving these codemods? Do they actually go and update the usages themselves or do we do it? 

Also, is there any other practise we do or would like to introduce to encourage teams switching using the new components?